### PR TITLE
Change values for 'templating_detected' from `<engine>` to `<engine>:…

### PR DESCRIPTION
**Pull request by wxtim**
sibbling: https://github.com/cylc/cylc-rose/pull/30

…suite.rc`

to allow for changes to the output of Cylc-Rose, which should be able
to include 'template variables', as well as 'jinja2:suite.rc' or
'empy:suite.rc'.

These changes partially address [Cylc Rose PR25](https://github.com/cylc/cylc-rose/issues/25), alongside [Cylc Rose PR 30](https://github.com/cylc/cylc-rose/pull/30)

**Requirements check-list**
- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
- [x] Contains logically grouped changes (else tidy your branch by rebase).
- [x] Does not contain off-topic changes (use other PRs for other changes).

- [x] Does not need tests: Functional tests for functionality reliant on the rose-cylc plugin included with https://github.com/cylc/cylc-rose/pull/30

### CODE REVIEW

File: cylc/flow/parsec/fileparse.py [hunk 1]
```diff
@@ -337,7 +337,7 @@ def read_and_proc(fpath, template_vars=None, viewcfg=None, asedit=False):
     # process with EmPy
     if do_empy:
         if (
-            extra_vars['templating_detected'] == 'empy' and
+            extra_vars['templating_detected'] == 'empy:suite.rc' and
```
Comment (hjoliver) [inline cylc/flow/parsec/fileparse.py:340]:
```suggestion
            extra_vars['templating_detected'] == 'empy:flow.cylc' and
```
(should this be changed at both ends?)

File: cylc/flow/parsec/fileparse.py [hunk 2]
```diff
@@ -362,7 +362,7 @@ def read_and_proc(fpath, template_vars=None, viewcfg=None, asedit=False):
     # process with Jinja2
     if do_jinja2:
         if (
-            extra_vars['templating_detected'] == 'jinja2' and
+            extra_vars['templating_detected'] == 'jinja2:suite.rc' and
```
Comment (hjoliver) [inline cylc/flow/parsec/fileparse.py:365]:
```suggestion
            extra_vars['templating_detected'] == 'jinja2:flow.cylc' and
```

(ditto)

File: cylc/flow/parsec/fileparse.py (hunk 1 above)
Comment (wxtim) [reply cylc/flow/parsec/fileparse.py:340]: No. The idea is that the old syntax is maintained for legacy purposes 🐬 , but that the Cylc8 way of doing things is to use 'template variables'. Oliver and I had a very brief discussion of this issue at https://github.com/cylc/cylc-rose/issues/25.

Comment (hjoliver) [reply cylc/flow/parsec/fileparse.py:340]: Ah, got it, thanks.

**Review by hjoliver — APPROVED** [review]

**Comment by wxtim** [comment]
Not required. :)

**Pull request closed without merging**